<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from hopper_v0.urdf.xacro           | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="hopper">
  <material name="red">
    <color rgba="1 0 0 1"/>
  </material>
  <material name="green">
    <color rgba="0 1 0 1"/>
  </material>
  <material name="blue">
    <color rgba="0 0 1 1"/>
  </material>
  <link name="world"/>
  <link name="torso">
    <visual>
      <geometry>
        <cylinder length="0.4" radius="0.05"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.4" radius="0.05"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.14159"/>
      <inertia ixx="0.0438513604167" ixy="0.0" ixz="0.0" iyy="0.0438513604167" iyz="0.0" izz="0.000654497916667"/>
    </inertial>
  </link>
  <gazebo reference="torso">
    <material>Gazebo/Red</material>
  </gazebo>
  <!-- We contrain the hopper to stay on the xz plane -->
  <!-- Gazebo does not support a planar joint, so we use two prismatic joints chained together -->
  <!-- We must also allow the torso link to rotate around the y axis, we use a continuous joint -->
  <link name="mid">
    <!-- Link used to chain the two prismatic joints together -->
    <inertial>
      <!-- gazebo requires inertia to be specified -->
      <mass value="0.0001"/>
      <!-- gazebo requires mass to be specified -->
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
  </link>
  <link name="mid2">
    <!-- link used between the second prismatic joint and the continuous joint -->
    <inertial>
      <mass value="0.0001"/>
      <inertia ixx="0.0" ixy="0.0" ixz="0.0" iyy="0.0" iyz="0.0" izz="0.0"/>
    </inertial>
  </link>
  <joint name="world_to_mid" type="prismatic">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis rpy="0 0 0" xyz="1 0 0"/>
    <parent link="world"/>
    <child link="mid"/>
    <limit effort="10000" lower="-1" upper="1000" velocity="10000"/>
  </joint>
  <joint name="mid_to_mid2" type="prismatic">
    <origin rpy="0 0 0" xyz="0 0 1.23"/>
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <parent link="mid"/>
    <child link="mid2"/>
    <limit effort="10000" lower="-10" upper="1000" velocity="10000"/>
  </joint>
  <joint name="mid2_to_torso" type="continuous">
    <origin rpy="0 0 0" xyz="0.0 0.0 0"/>
    <!-- joint pose in the parent frame-->
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <parent link="mid2"/>
    <child link="torso"/>
  </joint>
  <link name="thigh">
    <visual>
      <geometry>
        <cylinder length="0.45" radius="0.05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.225"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.45" radius="0.05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.225"/>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.225"/>
      <mass value="3.53428875"/>
      <inertia ixx="0.061850053125" ixy="0.0" ixz="0.0" iyy="0.061850053125" iyz="0.0" izz="0.00073631015625"/>
    </inertial>
  </link>
  <gazebo reference="thigh">
    <material>Gazebo/Green</material>
  </gazebo>
  <joint name="torso_to_thigh" type="revolute">
    <origin rpy="0 0 0" xyz="0.0 0.0 -0.2"/>
    <!-- joint pose in the parent frame-->
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <parent link="torso"/>
    <child link="thigh"/>
    <limit effort="1000" lower="0.0" upper="2.617991667" velocity="10000"/>
  </joint>
  <transmission name="torso_to_thigh_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <!-- Use a simple reducer transmission -->
    <joint name="torso_to_thigh">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <!-- Use an effort interface to control this joint (which is controlled through the actuator specified next) -->
    </joint>
    <actuator name="torso_to_thigh_actuator">
      <mechanicalReduction>200</mechanicalReduction>
      <!-- gear reduction for the joint -->
    </actuator>
  </transmission>
  <link name="leg">
    <visual>
      <geometry>
        <cylinder length="0.5" radius="0.04"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.25"/>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.5" radius="0.04"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.25"/>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 -0.25"/>
      <mass value="2.513272"/>
      <inertia ixx="0.0533651421333" ixy="0.0" ixz="0.0" iyy="0.0533651421333" iyz="0.0" izz="0.000335102933333"/>
    </inertial>
  </link>
  <gazebo reference="leg">
    <material>Gazebo/Blue</material>
  </gazebo>
  <joint name="thigh_to_leg" type="revolute">
    <origin rpy="0 0 0" xyz="0.0 0.0 -0.45"/>
    <parent link="thigh"/>
    <child link="leg"/>
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <limit effort="1000" lower="0.0" upper="2.617991667" velocity="10000"/>
  </joint>
  <transmission name="thigh_to_leg_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <!-- Use a simple reducer transmission -->
    <joint name="thigh_to_leg">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <!-- Use an effort interface to control this joint (which is controlled through the actuator specified next) -->
    </joint>
    <actuator name="thigh_to_leg_actuator">
      <mechanicalReduction>200</mechanicalReduction>
      <!-- gear reduction for the joint -->
    </actuator>
  </transmission>
  <link name="foot">
    <contact>
      <!-- PyBullet's way of specifying friction -->
      <lateral_friction value="1"/>
    </contact>
    <visual>
      <origin rpy="0 1.570795 0" xyz="0.065 0 0"/>
      <geometry>
        <cylinder length="0.39" radius="0.06"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0 1.570795 0" xyz="0.26 0 0"/>
      <geometry>
        <sphere radius="0.06"/>
      </geometry>
    </visual>
    <visual>
      <origin rpy="0 1.570795 0" xyz="-0.13 0 0"/>
      <geometry>
        <sphere radius="0.06"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 1.570795 0" xyz="0.065 0 0"/>
      <geometry>
        <cylinder length="0.39" radius="0.06"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 1.570795 0" xyz="0.26 0 0"/>
      <geometry>
        <sphere radius="0.06"/>
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 1.570795 0" xyz="-0.13 0 0"/>
      <geometry>
        <sphere radius="0.06"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 1.570795 0" xyz="0.065 0 0"/>
      <mass value="4.41079236"/>
      <inertia ixx="0.059876506287" ixy="0.0" ixz="0.0" iyy="0.059876506287" iyz="0.0" izz="0.001323237708"/>
    </inertial>
  </link>
  <gazebo reference="foot">
    <mu1 value="1"/>
    <mu2 value="1"/>
    <material>Gazebo/Grey</material>
  </gazebo>
  <joint name="leg_to_foot" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 -0.5"/>
    <parent link="leg"/>
    <child link="foot"/>
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <limit effort="1000" lower="-0.7853975" upper="0.7853975" velocity="10000"/>
  </joint>
  <transmission name="leg_to_foot_tran">
    <type>transmission_interface/SimpleTransmission</type>
    <!-- Use a simple reducer transmission -->
    <joint name="leg_to_foot">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <!-- Use an effort interface to control this joint (which is controlled through the actuator specified next) -->
    </joint>
    <actuator name="leg_to_foot_actuator">
      <mechanicalReduction>200</mechanicalReduction>
      <!-- gear reduction for the joint -->
    </actuator>
  </transmission>
</robot>

